core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/Envelope.java

```diff
@@ -43,7 +45,7 @@ private Envelope(EnvelopePart[] parts) {
         for (int i = 0; i < parts.length - 1; i++) {
             if (parts[i].getEndPos() != parts[i + 1].getBeginPos())
                 throw new OSRDError(ErrorType.EnvelopePartsNotContiguous);
-            if (parts[i].getEndSpeed() != parts[i + 1].getBeginSpeed())
+            if (!areSpeedsEqual(parts[i].getEndSpeed(), parts[i + 1].getBeginSpeed()))
```

Comment:
This is actually one of the few places where we can reasonably check for equality with no tolerance, the speed values at transitions should be copied without arithmetic